lib/sles4sap/console_redirection.pm

```diff
@@ -215,7 +215,11 @@ sub connect_target_to_serial {
     croak "OpenQA variable BASE_VM_ID undefined. Run 'redirection_init()' first" unless get_var('BASE_VM_ID');
     croak "IP address '$args{destination_ip}' is not valid." unless grep(/^$RE{net}{IPv4}$/, $args{destination_ip});
     croak 'Global variable "$serialdev" undefined' unless $serialdev;
-    croak "Console is already redirected to:" . script_output('hostname', quiet => 1) if check_serial_redirection();
```

Comment:
In case the test module suddenly fails, console redirection will not be terminated. 
With croak in place, post fail hook fails as well and cleanup won't be triggered.